core/app/models/workarea/bulk_action/product_edit.rb

```diff
@@ -52,13 +58,27 @@ def apply_pricing!(product)
 
         pricing_skus(product).each do |sku|
           sku_pricing = pricing.dup
-          prices = sku_pricing.delete('prices') unless sku.prices.blank?
+          prices = sku_pricing.delete('prices')
           sku.update_attributes!(sku_pricing)
 
-          if prices.present?
-            sku.prices.select(&:generic?).each do |price|
-              price.update_attributes!(prices.first)
+          if prices.present? && sku.prices.present?
+            sku.prices.each do |price|
+              updates = prices.each_with_object({}) do |(type, value), params|
+                next unless value['amount'].present?
+
+                if price.generic? || change_non_generic_price?(value)
+                  params[type] = amount_from(value, price[type])
+                end
+              end
+
+              price.update_attributes!(updates)
             end
+          elsif prices.present?
```

Comment:
I'm not sure this makes sense, if you're adjusting by a percentage I think you'd be pretty surprised a new price gets created. Let's remove this.